api/v1/common_types.go

```diff
@@ -491,3 +498,21 @@ const (
 	// downscale strategy targeting nodes which are least busy in terms of # flowfiles in queues.
 	LeastBusyClusterDownscaleStrategy ClusterScalingStrategy = "leastbusy"
 )
+
+// Change the list to {"controller-service", "funnel", "input-port","output-port", "processor", "process-group"} when all the type are available
+// +kubebuilder:validation:Enum={"process-group"}
+type ResourceType string
+
+const (
+	ResourceProcessGroup ResourceType = "process-group"
+	// ResourceProcessor          NiFiResourceType = "processor"
+	// ResourceInputPort          NiFiResourceType = "input-port"
+	// ResourceOutputPort         NiFiResourceType = "output-port"
+	// ResourceFunnel             NiFiResourceType = "funnel"
+	// ResourceControllerService  NiFiResourceType = "controller-service"
```

Comment:
```suggestion
	ResourceProcessor          NiFiResourceType = "processor"
	ResourceInputPort          NiFiResourceType = "input-port"
	ResourceOutputPort         NiFiResourceType = "output-port"
	ResourceFunnel             NiFiResourceType = "funnel"
	ResourceControllerService  NiFiResourceType = "controller-service"
```

You can uncomment them, as they are not available in the CRD as they aren't in the kubebuilder `Enum` validation.